### PR TITLE
Improving getOAuthUrl function and Create accountCreation function

### PR DESCRIPTION
**Pull request by bertrand-deriv**
### Improving getOAuthUrl function and Create accountCreation function
- Improved getOAuthUrl function to handle unauthorized account login
- Create accountCreation function to create real account automatically

### CODE REVIEW

File: cypress/support/common.js [hunk 1]
```diff
@@ -1,135 +1,185 @@
...
+        csrf_token: csrfToken
+      },
       headers: {
         'Content-Type': 'application/x-www-form-urlencoded',
-        'Origin': 'https://oauth.deriv.com'
+        'Origin': 'https://oauth.deriv.com',
+        'Cookie': cookie,
+        'csrf_token': csrfToken
       }
     }).then((response) => {
```
Comment (abuzar-deriv) [inline cypress/support/common.js:116]: Is it possible to create a function call for this? Code is getting duplicated here and _line 140-156_

File: cypress/support/common.js [hunk 2]
```diff
@@ -1,135 +1,185 @@
...
+                password: loginPassword,
+                login: 'Log in',
+                csrf_token: csrfToken
+              },
+              headers: {
+                'Content-Type': 'application/x-www-form-urlencoded',
+                'Origin': 'https://oauth.deriv.com',
+                'Cookie': cookie,
+                'csrf_token': csrfToken
+              }
```
Comment (abuzar-deriv) [inline cypress/support/common.js:156]: Refer to above comment.

File: cypress/support/accountCreation.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    await browser.close();
+  }
+})
+
+const createAccountVirtual = (async () => {
+  try {
+      const response = await api.basic.newAccountVirtual({
+        "new_account_virtual": 1,
+        "type": "trading",
+        "client_password": "Abcd1234",
```
Comment (abuzar-deriv) [inline cypress/support/accountCreation.js:85]: Hard-coded value. Please pass it through env variables.

File: cypress/support/accountCreation.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
+const { chromium } = require('playwright');
+require('dotenv').config()
+const WebSocket = require('ws');
+const DerivAPI = require('@deriv/deriv-api/dist/DerivAPI');
+const app_id = process.env.APP_ID
+const websocketURL = process.env.WEBSOCKET_URL
```
Comment (bilalmajid-deriv) [inline cypress/support/accountCreation.js:6]: I see we are not using cypress here but when running the test we actually set the value for appId depending on the scenario we are testing.

Shouldn't we be setting this up either conditionally or by getting the value dynamically based on which kind of  account needs to be created?

Comment (abuzar-deriv) [inline cypress/support/accountCreation.js:6]: Shouldn't these be defined inside config file and then called using Cypress.env?

File: cypress/support/accountCreation.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    // Add HTTP credentials for basic authentication
+    httpCredentials: {
+      username: process.env.EMAIL,
+      password: process.env.PASSWORD,
+    },
+  });
+  const page = await context.newPage();
+
+  // Navigate to the /events page
+  await page.goto(`${process.env.BASIC_AUTH_URL}/events`);
```
Comment (abuzar-deriv) [inline cypress/support/accountCreation.js:48]: Shouldn't these be defined inside config file and then called using Cypress.env?

**Review by bilalmajid-deriv — COMMENTED** [review]
For the createAccountReal and createAccountVirtual Function can we not make it parametrized?

so as to make custom tailored made accounts in accordance to the test being executed

File: cypress/support/common.js (hunk 1 above)
Comment (bertrand-deriv) [reply cypress/support/common.js:116]: Yeah...That's right @abuzar-deriv 
let me do it

Comment (bertrand-deriv) [reply cypress/support/common.js]: @abuzar-deriv  I just found that creating a separate for making such call would bring complexity in the code, rather I created a loginRequestPayload object variable that holds payload for such call, at least we don't repeat it twice

File: cypress/support/accountCreation.js (hunk 3 above)
Comment (bertrand-deriv) [reply cypress/support/accountCreation.js]: Values that are here like new_account_virtual and type won't change and are not sensitive, I thought putting them in .env would increase complexity for someone to use the function. Let me know what you think.

**Comment by bertrand-deriv** [comment]
This PR is no longer Valid.
@bilalmajid-deriv @abuzar-deriv  Would you please help to close it 🙏

**Comment by bertrand-deriv** [comment]
I'm going to close this PR since it is no longer valid

**Pull request closed without merging**